zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -243,8 +335,40 @@ private[codec] object EncoderDecoder                   {
       } else if (jsonDecoders.length == 1) {
         jsonDecoders(0)(body).map { result => inputs(0) = result }
       } else {
-        ZIO.foreachDiscard(jsonDecoders.zipWithIndex) { case (decoder, index) =>
-          decoder(body).map { result => inputs(index) = result }
+        val indexByName = flattened.content.zipWithIndex.map { case (codec, idx) =>
```

Comment:
This is computation that should happen outside the hot path. Move it as a field of the class so it can be done a single time, rather than once per request.